hera_mc/geo_sysdef.py

```diff
@@ -43,6 +44,63 @@
                         340, 341, 345, 346, 347, 348, 349]}
 
 
+def _get_region(this_ant):
+    """
+    Properly handle a single entry for finding a region.
+
+    If a full HERA part number is given and is for a ring, it will issue a
+    warning if the prefix given and the ring don't match.
+
+    Parameter
+    ---------
+    this_ant : str or int
+        Antenna number for which to check.
+    """
+    prefix = ''
+    if not isinstance(this_ant, int):
+        peeled = cm_utils.peel_key(this_ant, 'NPR')
+        this_ant = peeled[0]
+        prefix = peeled[1].upper()
+    for this_region, region_list in region.items():
+        if this_ant in region_list:
+            val = this_region[-1].upper()
+            break
+    else:
+        raise ValueError(f"{this_ant} is not valid antenna.")
+    if val in ['A', 'B'] and len(prefix) == 2:
+        if val != prefix[1]:
+            import warnings
+            warnings.warn(f"{prefix} does not match region {val}")
+    return val
+
+
+def ant_region(ants):
+    """
+    Provide the region of an antenna or list of antennas.
+
+    If a list, it will return a list, otherwise it will return a str.
+    Note that the regions are:  N, E, W, A, B for
+        North "tridrant"
+        East  "
+        West  "
+        A ring (inner)
+        B ring (outer)
+
+    Parameter
+    ---------
+    ants : str, int or list
+            Antenna number or HERA part number of antenna.
+    """
+    return_as = 'list'
+    if not isinstance(ants, list):
```

Comment:
You might want to also allow the input to be tuples or numpy arrays.

Reply:
I made it more flexible to allow other list-y options, using TypeError vs ValueError.  Did similar in the _get_region module.